tests/connection/test_connection_service.py

```diff
@@ -827,7 +794,7 @@ def test_connect_with_access_token(self):
         })
 
         # Set up the mock connection for psycopg2's connect method to return
-        mock_connection = MockConnection(dsn_parameters={
+        mock_connection = MockPsycopgConnection(dsn_parameters={
```

Comment:
Do we need this here? We have already defined the self.mock_connection in setup. We can just use that